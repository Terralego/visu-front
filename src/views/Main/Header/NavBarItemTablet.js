import React from 'react';
import PropTypes from 'prop-types';

import withEnv from '../../../config/withEnv';
import HeaderLink from './HeaderLink';
import HeaderButton from './HeaderButton';

export const NavBarItemTablet = ({
  id,
  iconPath,
  env: { VIEW_ROOT_PATH },
  ...item
}) => (
  <HeaderLink
    href={(item.href || '').replace('{{VIEW_ROOT_PATH}}', VIEW_ROOT_PATH)}
    onClick={item.onClick}
    data-link-id={id}
  >
    <p className="header-button-label">{item.label}</p>
    <HeaderButton
      id={id}
      iconPath={iconPath}
      alt={item.label}
    />
  </HeaderLink>
);

NavBarItemTablet.propTypes = {
  env: PropTypes.shape({
    VIEW_ROOT_PATH: PropTypes.string,
  }).isRequired,
  id: PropTypes.string.isRequired,
  iconPath: PropTypes.string,
  item: PropTypes.shape({
    href: PropTypes.string,
    label: PropTypes.string,
  }),
};

NavBarItemTablet.defaultProps = {
  iconPath: '',
  item: {
    href: '',
    label: '',
    onClick () {},
  },
};

export default withEnv(NavBarItemTablet);
