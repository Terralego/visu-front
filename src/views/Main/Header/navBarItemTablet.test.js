import React from 'react';
import renderer from 'react-test-renderer';
import { NavBarItemTablet } from './NavBarItemTablet';

jest.mock('./HeaderLink', () => () => 'HeaderLink');
jest.mock('./HeaderButton', () => () => 'HeaderButton');

it('should render correctly the component', () => {
  const tree = renderer.create((
    <NavBarItemTablet env={{ VIEW_ROOT_PATH: 'visualiser' }} id="test" />
  )).toJSON();
  expect(tree).toMatchSnapshot();
});
