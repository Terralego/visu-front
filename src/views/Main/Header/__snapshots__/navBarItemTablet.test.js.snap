// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly the component 1`] = `"HeaderLink"`;
